lua/nvim-treesitter/pairs.lua

```diff
@@ -78,7 +78,7 @@ function M.delete_balanced(buf)
     vim.lsp.util.apply_text_edits({
       { range = ts_utils.node_to_lsp_range(self), newText = ''},
       { range = ts_utils.node_to_lsp_range(partner), newText = ''},
-    }, buf)
+    }, buf, "utf-16")
   end
```

Comment:
I think the offsets reported by Tree-sitter are actually bytes and not utf-offsets. But we might use that as a quick fix